### DISCUSSION TITLE
Is this a bug? Imported constant is not defined while destructuring object

### DISCUSSION THREAD

**Discussion by inakiabt** [vitest-dev/vitest#1789] [Q&A] [answered]
https://stackblitz.com/edit/vitest-dev-vitest-eg6hlk?file=test/helper2.js

there are 3 tests. all of them tests a simple function that uses an imported constant to destruct an object and return the result.
- for test 1 and 2, the constant (NAME) seems to be "not defined" to vitest.
- for test 3, I use a different the constant (n) that seems to be defined but it doesn't have the correct value when running the test.

the weirdest thing is that I'm printing the constant value without issues
it is driving me crazy. please. H.E.L.P 🙂

**Comment by sheremet-va** [comment, marked as answer]
This is Vite transform error. If you open Module Graph page in Vitest UI, you can inspect what the file is transformed into. I suggest opening an issue in Vitest repo with your reproduction.

**Reply by inakiabt** [reply]
It is indeed a transform error:
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/606852/182933147-cc8b5056-7292-4652-9ce3-46473a406bf3.png">

I'll open an issue. Thanks!
